queries.sql

```diff
@@ -128,4 +128,92 @@ from owners o
 left join animals a on o.owner_id = a.owner_id
 group by o.full_name
 order by count(a.owner_id) desc
-limit 1;
\ No newline at end of file
+limit 1;
+
+/* Last animal seen by William Tatcher*/
+select a.animal_name
+from visits v
+inner join vets vt on v.vet_id = vt.vet_id
+inner join animals a on v.animal_id = a.animal_id
+where vt.vet_name = 'William Tatcher'
+order by v.visit_date desc
+limit 1;
+
+/*Different animals seen by Stephanie Mendez*/
+select count (distinct v.animal_id)
+from visits v
+inner join vets vt on v.vet_id =vt.vet_id
+where vt.vet_name = 'Stephanie Mendez';
+
+/* All vets and their specialities, including vets with no specialities*/
+select vt.vet_name, coalesce(array_agg(sp.species_id order by sp.species_id), '{}') as specialties
+from vets vt
+left join specializations sp on vt.vet_id = sp.vet_id
+left join species s on sp.species_id = s.species_id
+group by vt.vet_name;
+
+/* All animals that visited Stephanie Mendez between April 1st and August 30th 2020*/
+select a.animal_name
+from visits v
+inner join vets vt on v.vet_id = vt.vet_id
+inner join animals a on v.animal_id = a.animal_id
+where vt.vet_name = 'Stephanie Mendez'
+	and v.visit_date >= '2020-04-01'
+	and v.visit_date <= '2020-08-30';
+
+
+/* Animal that has the most visits to vets*/
+select a.animal_name
+from (
+    select animal_id,count(*) as visit_count
+    from visits 
+    group by animal_id
+    order by visit_count desc
+    limit 1
+) most_visited
+join animals a on most_visited.animal_id = a.animal_id;
+
+/* Maisy Smith's first visit */
+select a.animal_name, v.visit_date
+from visits v
+inner join vets vt on v.vet_id = vt.vet_id
+inner join animals a on v.animal_id = a.animal_id
+where vt.vet_name = 'Maisy Smith'
+order by v.visit_date asc
+limit 1;
+
+/*Details for the most recent visit: animal information, vet information, and date of visit*/
+select s.species_name as recommended_speciality
+from(
+    select an.species_id, count(*) as visit_count
+    from visits v
+    inner join animals an on v.animal_id = an.animal_id
+	inner join vets vt on v.vet_id = vt.vet_id
+	where vt.vet_name = 'Maisy Smith'
+	group by an.species_id
+	order by visit_count desc
+	limit 1
+) most_visited_species
+join species s on most_visited_species.species_id = s.species_id;
```

Comment:
- [ ] This query was duplicated twice. you're meant to search `Details for most recent visit: animal information, vet information, and date of visit`, however, you did a query similar to the last query. Kindly look at the sample below

### Sample:
```
  SELECT animals.*, vets.*, vistis.visit_date
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  ORDER BY visits.visit_date DESC
  LIMIT 1;
  ```
- Note: The above is just a sample. your query should make sure to give the right results

Reply:
Most recent visit ;animal information, vet information and visit date
![most recent visit animal info vet info date of visit](https://github.com/iankiema/Vet-Clinic-Database/assets/112932127/d16843ba-4240-4d5b-8236-c2c673d99b2b)